src/wallet/rpcwallet.cpp

```diff
@@ -4555,6 +4555,30 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
             throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, duplicated address: ") + addrStr);
         };
 
+        std::optional<libzcash::UnifiedAddress> ua = std::nullopt;
+        if (std::holds_alternative<libzcash::UnifiedAddress>(decoded.value())) {
+            ua = std::get<libzcash::UnifiedAddress>(decoded.value());
+
+            for (const auto& receiver : *ua) {
+                std::visit(match {
+                    [&](const libzcash::SaplingPaymentAddress& addr) {
+                        addrStr = keyIO.EncodePaymentAddress(addr);
+                    },
+                    [&](const CScriptID& addr) {
+                        addrStr = keyIO.EncodePaymentAddress(addr);
+                    },
+                    [&](const CKeyID& addr) {
+                        addrStr = keyIO.EncodePaymentAddress(addr);
+                    },
+                    [](auto rest) {},
+                }, receiver);
+ 
+                if (!addrStrings.insert(addrStr).second) {
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, duplicated address: ") + addrStr);
+                };
```

Comment:
If any `receiver` matches the `auto rest` case, then the fact that `addrStr` is being reused here means that the previous value of `addrStr` will be added twice to `addrStrings`, triggering the error when it shouldn't be. At a minimum, this should be:
```suggestion
                std::optional<std::string> receiverStr;
                std::visit(match {
                    [&](const libzcash::SaplingPaymentAddress& addr) {
                        receiverStr = keyIO.EncodePaymentAddress(addr);
                    },
                    [&](const CScriptID& addr) {
                        receiverStr = keyIO.EncodePaymentAddress(addr);
                    },
                    [&](const CKeyID& addr) {
                        receiverStr = keyIO.EncodePaymentAddress(addr);
                    },
                    [](auto rest) {},
                }, receiver);
 
                if (receiverStr.has_value()) {
                    if (!addrStrings.insert(receiverStr.value()).second) {
                        throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, duplicated address: ") + addrStr);
                    }
                }
```
However, this means we aren't checking other receiver kinds like Orchard. I'd prefer that the `auto rest` case be removed, and we explicitly handle all receiver kinds (using single-receiver UAs as the stand-in for `receiverStr` in those cases).

Reply:
Aren't Orchard receivers always guaranteed to be unique since they can only be created by a `z_getaddressforaccount,` meaning we don't have to check them in this case? If the UA itself is duplicated, the uniqueness check catches it now.

Reply:
No; you could have two distinct unified addresses, one with a subset of the receivers of the other.

Reply:
When we were initially talking about this PR, I had not anticipated that you'd be checking string conversions; instead, I had thought that you'd perform the uniqueness check on the values of the `SendManyRecipient` recipient addresses. https://github.com/zcash/zcash/blob/master/src/wallet/rpcwallet.cpp#L4520

Reply:
How could you get two distint UAs that share some subset of receivers? I thought z_getaddressforaccount would create new addresses for each the pools you specify or the default set of pools if not specified.

Reply:
> When we were initially talking about this PR, I had not anticipated that you'd be checking string conversions

The logic that was already present checked string conversions so I modeled after that. Wouldn't those recipient addresses would only contain the UA string, not the individual UA receivers that comprise a UA?

Reply:
So, you can make two calls:
`z_getaddressforaccount 0 ['transparent', 'orchard']`
`z_getaddressforaccount 0 ['sapling', 'orchard']`
and thereby have two distinct UAs. Not to mention things like situations where people reuse the same mnemonic seed across multiple wallets that have different policies.

Reply:
Here's a suggestion for how I'd write this: https://github.com/nuttycom/zcash/commit/0a34735bf8ccad8ce1b34f7f6363572bdd769316

Reply:
@nuttycom I left a question in the gist.

Reply:
Comments addressed in latest commit.